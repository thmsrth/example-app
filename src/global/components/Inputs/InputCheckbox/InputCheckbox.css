.InputCheckbox--alert .InputCheckbox__options {
  border: 1px solid var(--form-input-alert-message-color);
}
