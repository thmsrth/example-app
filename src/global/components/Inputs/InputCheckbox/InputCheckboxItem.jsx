import React from 'react';
import PropTypes from 'prop-types';

import { bemCls } from '../../../utils/ClassNameHelpers';

const InputCheckboxItem = ({
  className, name, label, value, onChange, onBlur, disabled,
}) => (
  <div className={`InputCheckboxItem ${className} ${
    bemCls(`InputCheckboxItem ${className}`, '--checked', value)}`}
  >
    <label
      key={name}
      className={`${bemCls(`InputCheckboxItem ${className}`, '__label')} ${
        bemCls(`InputCheckboxItem ${className}`, `__${name}__label`)}`}
    >
      <input
        type="checkbox"
        className={`${bemCls(`InputCheckboxItem ${className}`, '__input')} ${
          bemCls(`InputCheckboxItem ${className}`, `__${name}__input`)} ${
          bemCls(`InputCheckboxItem ${className}`, '__input--disabled', disabled)}`}
        name={name}
        value={value}
        checked={value}
        onChange={onChange}
        onBlur={onBlur}
      />
      {label}
    </label>
  </div>
);

export default InputCheckboxItem;

InputCheckboxItem.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  className: PropTypes.string,
  value: PropTypes.bool,
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
};

InputCheckboxItem.defaultProps = {
  label: '',
  className: '',
  value: false,
  disabled: false,
  onChange: () => {},
  onBlur: () => {},
};
