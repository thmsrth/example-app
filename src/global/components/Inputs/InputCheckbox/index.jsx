import React from 'react';

import InputCheckbox from './InputCheckbox';

const InputCheckboxWrapper = props => (
  <InputCheckbox {...props} />
);

export default InputCheckboxWrapper;
