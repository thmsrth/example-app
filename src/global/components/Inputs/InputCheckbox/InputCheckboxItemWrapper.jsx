import React, { Component } from 'react';

import PropTypes from 'prop-types';
import validationKeysType from '../../Form/types/validationKeys';
import validationKeysDefaults from '../../Form/defaults/validationKeys';

import InputCheckboxItem from './InputCheckboxItem';

class InputCheckboxItemWrapper extends Component {
  constructor() {
    super();
    this.handleChange = this.handleChange.bind(this);
    this.handleBlur = this.handleBlur.bind(this);
  }

  handleChange(event) {
    const {
      name, value, option, onChange,
    } = this.props;

    const newValue = {
      ...value,
      [option]: event.target.checked,
    };
    onChange(name, newValue);
  }

  handleBlur() {
    const {
      name, validationKeys, onBlur,
    } = this.props;
    const options = { type: 'checkbox' };

    onBlur(name, validationKeys, options);
  }

  render() {
    const { value, option, ...otherProps } = this.props;
    return (
      <InputCheckboxItem
        {...otherProps}
        value={value[option]}
        onChange={this.handleChange}
        onBlur={this.handleBlur}
      />
    );
  }
}

export default InputCheckboxItemWrapper;

InputCheckboxItemWrapper.propTypes = {
  value: PropTypes.objectOf(PropTypes.bool),
  name: PropTypes.string,
  validationKeys: validationKeysType,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  option: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.number,
    PropTypes.string,
  ]),
};

InputCheckboxItemWrapper.defaultProps = {
  value: null,
  name: '',
  validationKeys: validationKeysDefaults,
  option: '',
  onChange: () => {},
  onBlur: () => {},
};
