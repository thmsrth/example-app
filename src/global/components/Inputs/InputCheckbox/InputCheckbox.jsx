import React from 'react';

import PropTypes from 'prop-types';
import validationKeysType from '../../Form/types/validationKeys';
import validationKeysDefaults from '../../Form/defaults/validationKeys';

import { bemCls } from '../../../utils/ClassNameHelpers';
import InputCheckboxItemWrapper from './InputCheckboxItemWrapper';

import './InputCheckbox.css';

const InputCheckbox = ({
  label, name, options, value, onChange, className, validationKeys, disabled, ...otherProps
}) => {
  const {
    required,
    validationAlert,
    warningAlert,
  } = validationKeys || InputCheckbox.defaultProps.validationKeys;

  return (
    <ul className={`InputCheckbox ${className} ${
      bemCls(`InputCheckbox ${className}`, `__${name}`)} ${
      bemCls(`InputCheckbox ${className}`, '__required', required)} ${
      bemCls(`InputCheckbox ${className}`, '--alert', validationAlert)}`}
    >
      <div
        className={`${bemCls(`InputCheckbox ${className}`, '__label')} ${
          bemCls(`InputCheckbox ${className}`, '__label--disabled', disabled)}`}
      >
        {`${label}${required ? '*' : ''}`}
        <div className={`${bemCls(`InputCheckbox ${className}`, '__alertMsg')} ${
          bemCls(`InputCheckbox ${className}`, `__${name}__alertMsg`)}`}
        >
          {validationAlert}
        </div>
        <div className={bemCls(`InputCheckbox ${className}`, '__options')}>
          {options.map(option => (
            <li
              key={option.value}
              className={`InputCheckbox ${className} ${
                bemCls(`InputCheckbox ${className}`, '--active', value)}`}
            >
              <div className={bemCls(`InputCheckbox ${className}`, '__container')}>
                <InputCheckboxItemWrapper
                  {...otherProps}
                  type="checkbox"
                  className={bemCls(`InputCheckbox ${className}`, '__item')}
                  name={name}
                  label={option.label}
                  value={value}
                  onChange={onChange}
                  option={option.value}
                  validationKeys={validationKeys}
                  disabled={disabled}
                />
              </div>
            </li>
          ))}
        </div>
        <div className={`${bemCls(`InputCheckbox ${className}`, '__warningMsg')} ${
          bemCls(`InputCheckbox ${className}`, `__${name}__warningMsg`)}`}
        >
          {warningAlert}
        </div>
      </div>
    </ul>
  );
};

export default InputCheckbox;

InputCheckbox.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape({
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
    label: PropTypes.string,
  })),
  value: PropTypes.objectOf(PropTypes.bool),
  onChange: PropTypes.func,
  className: PropTypes.string,
  validationKeys: validationKeysType,
  disabled: PropTypes.bool,
};

InputCheckbox.defaultProps = {
  name: '',
  label: '',
  options: [],
  value: null,
  onChange: () => {},
  className: '',
  validationKeys: validationKeysDefaults,
  disabled: false,
};
