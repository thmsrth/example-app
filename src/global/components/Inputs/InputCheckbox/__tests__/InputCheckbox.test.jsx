import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import InputCheckbox from '../InputCheckbox';

const props = {
  className: 'MyInputCheckbox',
  name: 'mycheckbox',
  label: 'My Input Label',
  validationKeys: {
    required: false,
    validationAlert: '',
    warningAlert: '',
  },
  onChange: () => {},
  options: [
    {
      label: 'Hi',
      value: 1,
    },
    {
      label: 'Ho',
      value: 2,
    },
  ],
};

describe('InputCheckbox', () => {
  it('should render the component correctly and return a false default value', () => {
    const tree = shallow(<InputCheckbox {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component correctly and value 1: true, 2: false', () => {
    const tree = shallow(<InputCheckbox {...props} value={{ 1: true, 2: false }} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component as a required input', () => {
    const tree = shallow(<InputCheckbox
      {...props}
      validationKeys={{ required: true }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with a validation alert message', () => {
    const tree = shallow(<InputCheckbox
      {...props}
      validationKeys={{ validationAlert: 'My validation alert message' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with a warning message', () => {
    const tree = shallow(<InputCheckbox
      {...props}
      validationKeys={{ warningAlert: 'My warning message' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
