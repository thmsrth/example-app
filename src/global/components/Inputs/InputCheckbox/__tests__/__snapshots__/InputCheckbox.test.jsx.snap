// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCheckbox should render the component as a required input 1`] = `
<ul
  className="InputCheckbox MyInputCheckbox InputCheckbox__mycheckbox MyInputCheckbox__mycheckbox InputCheckbox__required MyInputCheckbox__required InputCheckbox--alert MyInputCheckbox--alert"
>
  <div
    className="InputCheckbox__label MyInputCheckbox__label "
  >
    My Input Label*
    <div
      className="InputCheckbox__alertMsg MyInputCheckbox__alertMsg InputCheckbox__mycheckbox__alertMsg MyInputCheckbox__mycheckbox__alertMsg"
    />
    <div
      className="InputCheckbox__options MyInputCheckbox__options"
    >
      <li
        className="InputCheckbox MyInputCheckbox "
        key="1"
      >
        <div
          className="InputCheckbox__container MyInputCheckbox__container"
        >
          <InputCheckboxItemWrapper
            className="InputCheckbox__item MyInputCheckbox__item"
            disabled={false}
            label="Hi"
            name="mycheckbox"
            onBlur={[Function]}
            onChange={[Function]}
            option={1}
            type="checkbox"
            validationKeys={
              Object {
                "required": true,
              }
            }
            value={null}
          />
        </div>
      </li>
      <li
        className="InputCheckbox MyInputCheckbox "
        key="2"
      >
        <div
          className="InputCheckbox__container MyInputCheckbox__container"
        >
          <InputCheckboxItemWrapper
            className="InputCheckbox__item MyInputCheckbox__item"
            disabled={false}
            label="Ho"
            name="mycheckbox"
            onBlur={[Function]}
            onChange={[Function]}
            option={2}
            type="checkbox"
            validationKeys={
              Object {
                "required": true,
              }
            }
            value={null}
          />
        </div>
      </li>
    </div>
    <div
      className="InputCheckbox__warningMsg MyInputCheckbox__warningMsg InputCheckbox__mycheckbox__warningMsg MyInputCheckbox__mycheckbox__warningMsg"
    />
  </div>
</ul>
`;

exports[`InputCheckbox should render the component correctly and return a false default value 1`] = `
<ul
  className="InputCheckbox MyInputCheckbox InputCheckbox__mycheckbox MyInputCheckbox__mycheckbox  "
>
  <div
    className="InputCheckbox__label MyInputCheckbox__label "
  >
    My Input Label
    <div
      className="InputCheckbox__alertMsg MyInputCheckbox__alertMsg InputCheckbox__mycheckbox__alertMsg MyInputCheckbox__mycheckbox__alertMsg"
    />
    <div
      className="InputCheckbox__options MyInputCheckbox__options"
    >
      <li
        className="InputCheckbox MyInputCheckbox "
        key="1"
      >
        <div
          className="InputCheckbox__container MyInputCheckbox__container"
        >
          <InputCheckboxItemWrapper
            className="InputCheckbox__item MyInputCheckbox__item"
            disabled={false}
            label="Hi"
            name="mycheckbox"
            onBlur={[Function]}
            onChange={[Function]}
            option={1}
            type="checkbox"
            validationKeys={
              Object {
                "required": false,
                "validationAlert": "",
                "warningAlert": "",
              }
            }
            value={null}
          />
        </div>
      </li>
      <li
        className="InputCheckbox MyInputCheckbox "
        key="2"
      >
        <div
          className="InputCheckbox__container MyInputCheckbox__container"
        >
          <InputCheckboxItemWrapper
            className="InputCheckbox__item MyInputCheckbox__item"
            disabled={false}
            label="Ho"
            name="mycheckbox"
            onBlur={[Function]}
            onChange={[Function]}
            option={2}
            type="checkbox"
            validationKeys={
              Object {
                "required": false,
                "validationAlert": "",
                "warningAlert": "",
              }
            }
            value={null}
          />
        </div>
      </li>
    </div>
    <div
      className="InputCheckbox__warningMsg MyInputCheckbox__warningMsg InputCheckbox__mycheckbox__warningMsg MyInputCheckbox__mycheckbox__warningMsg"
    />
  </div>
</ul>
`;

exports[`InputCheckbox should render the component correctly and value 1: true, 2: false 1`] = `
<ul
  className="InputCheckbox MyInputCheckbox InputCheckbox__mycheckbox MyInputCheckbox__mycheckbox  "
>
  <div
    className="InputCheckbox__label MyInputCheckbox__label "
  >
    My Input Label
    <div
      className="InputCheckbox__alertMsg MyInputCheckbox__alertMsg InputCheckbox__mycheckbox__alertMsg MyInputCheckbox__mycheckbox__alertMsg"
    />
    <div
      className="InputCheckbox__options MyInputCheckbox__options"
    >
      <li
        className="InputCheckbox MyInputCheckbox InputCheckbox--active MyInputCheckbox--active"
        key="1"
      >
        <div
          className="InputCheckbox__container MyInputCheckbox__container"
        >
          <InputCheckboxItemWrapper
            className="InputCheckbox__item MyInputCheckbox__item"
            disabled={false}
            label="Hi"
            name="mycheckbox"
            onBlur={[Function]}
            onChange={[Function]}
            option={1}
            type="checkbox"
            validationKeys={
              Object {
                "required": false,
                "validationAlert": "",
                "warningAlert": "",
              }
            }
            value={
              Object {
                "1": true,
                "2": false,
              }
            }
          />
        </div>
      </li>
      <li
        className="InputCheckbox MyInputCheckbox InputCheckbox--active MyInputCheckbox--active"
        key="2"
      >
        <div
          className="InputCheckbox__container MyInputCheckbox__container"
        >
          <InputCheckboxItemWrapper
            className="InputCheckbox__item MyInputCheckbox__item"
            disabled={false}
            label="Ho"
            name="mycheckbox"
            onBlur={[Function]}
            onChange={[Function]}
            option={2}
            type="checkbox"
            validationKeys={
              Object {
                "required": false,
                "validationAlert": "",
                "warningAlert": "",
              }
            }
            value={
              Object {
                "1": true,
                "2": false,
              }
            }
          />
        </div>
      </li>
    </div>
    <div
      className="InputCheckbox__warningMsg MyInputCheckbox__warningMsg InputCheckbox__mycheckbox__warningMsg MyInputCheckbox__mycheckbox__warningMsg"
    />
  </div>
</ul>
`;

exports[`InputCheckbox should render the component with a validation alert message 1`] = `
<ul
  className="InputCheckbox MyInputCheckbox InputCheckbox__mycheckbox MyInputCheckbox__mycheckbox InputCheckbox__required MyInputCheckbox__required InputCheckbox--alert MyInputCheckbox--alert"
>
  <div
    className="InputCheckbox__label MyInputCheckbox__label "
  >
    My Input Label
    <div
      className="InputCheckbox__alertMsg MyInputCheckbox__alertMsg InputCheckbox__mycheckbox__alertMsg MyInputCheckbox__mycheckbox__alertMsg"
    >
      My validation alert message
    </div>
    <div
      className="InputCheckbox__options MyInputCheckbox__options"
    >
      <li
        className="InputCheckbox MyInputCheckbox "
        key="1"
      >
        <div
          className="InputCheckbox__container MyInputCheckbox__container"
        >
          <InputCheckboxItemWrapper
            className="InputCheckbox__item MyInputCheckbox__item"
            disabled={false}
            label="Hi"
            name="mycheckbox"
            onBlur={[Function]}
            onChange={[Function]}
            option={1}
            type="checkbox"
            validationKeys={
              Object {
                "validationAlert": "My validation alert message",
              }
            }
            value={null}
          />
        </div>
      </li>
      <li
        className="InputCheckbox MyInputCheckbox "
        key="2"
      >
        <div
          className="InputCheckbox__container MyInputCheckbox__container"
        >
          <InputCheckboxItemWrapper
            className="InputCheckbox__item MyInputCheckbox__item"
            disabled={false}
            label="Ho"
            name="mycheckbox"
            onBlur={[Function]}
            onChange={[Function]}
            option={2}
            type="checkbox"
            validationKeys={
              Object {
                "validationAlert": "My validation alert message",
              }
            }
            value={null}
          />
        </div>
      </li>
    </div>
    <div
      className="InputCheckbox__warningMsg MyInputCheckbox__warningMsg InputCheckbox__mycheckbox__warningMsg MyInputCheckbox__mycheckbox__warningMsg"
    />
  </div>
</ul>
`;

exports[`InputCheckbox should render the component with a warning message 1`] = `
<ul
  className="InputCheckbox MyInputCheckbox InputCheckbox__mycheckbox MyInputCheckbox__mycheckbox InputCheckbox__required MyInputCheckbox__required InputCheckbox--alert MyInputCheckbox--alert"
>
  <div
    className="InputCheckbox__label MyInputCheckbox__label "
  >
    My Input Label
    <div
      className="InputCheckbox__alertMsg MyInputCheckbox__alertMsg InputCheckbox__mycheckbox__alertMsg MyInputCheckbox__mycheckbox__alertMsg"
    />
    <div
      className="InputCheckbox__options MyInputCheckbox__options"
    >
      <li
        className="InputCheckbox MyInputCheckbox "
        key="1"
      >
        <div
          className="InputCheckbox__container MyInputCheckbox__container"
        >
          <InputCheckboxItemWrapper
            className="InputCheckbox__item MyInputCheckbox__item"
            disabled={false}
            label="Hi"
            name="mycheckbox"
            onBlur={[Function]}
            onChange={[Function]}
            option={1}
            type="checkbox"
            validationKeys={
              Object {
                "warningAlert": "My warning message",
              }
            }
            value={null}
          />
        </div>
      </li>
      <li
        className="InputCheckbox MyInputCheckbox "
        key="2"
      >
        <div
          className="InputCheckbox__container MyInputCheckbox__container"
        >
          <InputCheckboxItemWrapper
            className="InputCheckbox__item MyInputCheckbox__item"
            disabled={false}
            label="Ho"
            name="mycheckbox"
            onBlur={[Function]}
            onChange={[Function]}
            option={2}
            type="checkbox"
            validationKeys={
              Object {
                "warningAlert": "My warning message",
              }
            }
            value={null}
          />
        </div>
      </li>
    </div>
    <div
      className="InputCheckbox__warningMsg MyInputCheckbox__warningMsg InputCheckbox__mycheckbox__warningMsg MyInputCheckbox__mycheckbox__warningMsg"
    >
      My warning message
    </div>
  </div>
</ul>
`;
